functions/src/index.ts

```diff
@@ -37,203 +37,209 @@ export const initializeApp = configService.setConfig;
+  const distFolder = await _getDistFolder();
+  response.setHeader('Cache-Control', `public, max-age=600, s-maxage=3600, stale-while-revalidate=120`);
 
-    response.sendFile(join(distFolder, 'admin.html'));
-    return _registerHostname(request);
+  response.sendFile(join(distFolder, 'admin.html'));
+  return _registerHostname(request);
 });
 
 app.get('/manifest.json', (request, response) => {
-    response.status(404).send('Not implemented yet');
+  response.status(404).send('Not implemented yet');
 });
 
 app.get('/sitemap.xml', async (request, response) => {
-    const sitemap = await fileService.getSitemap();
-    response.setHeader('Cache-Control', getCacheHeader());
-    response.send(sitemap);
-    return null;
+  const sitemap = await fileService.getSitemap();
+  response.setHeader('Cache-Control', getCacheHeader());
+  response.send(sitemap);
+  return null;
 });
 
 app.get('/robots.txt', (request, response) => {
-    const robotsDefinition = ['User-agent: *', 'Disallow: /_/'];
-    response.send(robotsDefinition.join('\n'));
+  const robotsDefinition = ['User-agent: *', 'Disallow: /_/'];
+  response.send(robotsDefinition.join('\n'));
 });
 
 app.get('/favicon.ico', async (request, response) => {
-    response.setHeader('Content-Type', 'image/ico');
-    response.setHeader('Cache-Control', getCacheHeader());
-
-    const favicon = await fileService.getFavicon();
-    if (favicon) {
-        response.send(favicon);
-        return null;
-    }
+  response.setHeader('Content-Type', 'image/ico');
+  response.setHeader('Cache-Control', getCacheHeader());
 
-    const distFolder = await _getDistFolder();
-    response.sendFile(join(distFolder, 'favicon.ico'));
+  const favicon = await fileService.getFavicon();
+  if (favicon) {
+    response.send(favicon);
     return null;
+  }
+
+  const distFolder = await _getDistFolder();
+  response.sendFile(join(distFolder, 'favicon.ico'));
+  return null;
 });
 
 app.get('*', async (request, response) => {
-    console.log(`GET ${request.url}`);
-    const url = request.url.replace(/^\//, ''); // Remove leading slash
-
-    const context = await getPageContextByUrl(url);
-    if (!context) {
-        console.log(`[HTTP 404] page not found for: ${request.url}`);
-        return response.status(404).send(`Not found: ${request.url}`);
-    }
-
-    const html = await renderPage(context);
-    if (!html) {
-        console.error(`[HTTP 500] could not create template for: ${request.url}`);
-        return response.status(500).send('Could not create HTML template document');
-    }
-
+  console.log(`GET ${request.url}`);
+  const url = request.url.replace(/^\//, ''); // Remove leading slash
+
+  const context = await getPageContextByUrl(url);
+  if (!context) {
+    // If documenpage 404  doesn't set up
+    console.log(`[HTTP 404] page not found for: ${request.url}`);
+    return response.status(404).send(`Not found: ${request.url}`);
```

Comment:
Here is where we should fetch and render the `http404` template **if** it exists in the theme templats.

Reply:
Resolve on this commit https://github.com/oddbit/tanam/pull/213/commits/0160877a37bd499709a9872dd5c7ea445b482e96